src/StreamClient.cs

```diff
@@ -0,0 +1,212 @@
+﻿using Stream.Models;
+using Stream.Rest;
+using Stream.Utils;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace Stream
+{
+    /// <summary>
+    /// A factory-like class that can be used for basic operations and

```

Comment:
yeah. I hope it makes sense

Reply:
sorry, I should've expanded on what I meant. I'm not sure if from this class user perspective it works as a factory, You usually use factories to instantiate objects on demand and hide the instantiation complexity under the hood. 

In this case, you just use this as the main client to access more specific API Clients.

It would be a factory if you'd use it to create those clients and have methods like:
```
IUsers CreateUsersClient()
IReactions CreateReactionsClient()
etc.
```

Of course, we could say that it is a factory of requests, which is true, but if that's the interpretation then every sub-client like IUsers, IImages, etc. should be described as factory as well.

Reply:
okay, rephrased it